solutionbox/structured_data/datalab_solutions/structured_data/trainer/util.py

```diff
@@ -91,6 +105,26 @@ def end(self, session=None):
       _copy_all(self._additional_assets, assets_dir)
 
 
+# ==============================================================================
+# Reading the input csv files and parsing its output into tensors.
+# ==============================================================================
+
+def parse_example_tensor(examples, train_config):
+
+  # record_defaults are used by tf.decode_csv to insert defaults, and to infer
+  # the datatype. 
+  record_defaults = [[train_config['csv_defaults'][name]] 
```

Comment:
csv_defaults can be optional too? We can use default value based on schema. We can require no missing values if defaults are not provided?